platform/core/src/measurements/classes/MeasurementApi.js

```diff
@@ -203,13 +204,14 @@ export default class MeasurementApi {
 
   retrieveMeasurements(patientId, timepointIds) {
     const retrievalFn = configuration.dataExchange.retrieve;
+    const { server } = this.options;
     if (typeof retrievalFn !== 'function') {
       log.error('Measurement retrieval function has not been configured.');
       return;
     }
 
     return new Promise((resolve, reject) => {
-      retrievalFn(patientId, timepointIds).then(measurementData => {
+      retrievalFn({ patientId, timepointIds, server }).then(measurementData => {
```

Comment:
The `...` that sets this in configuration/dataExchange made it difficult to track down where `retrievalFn` is defined. Outside the scope of this PR -- just making a note for follow-up issue(s).

Reply:
Totally agree with you Danny.